### PR TITLE
fix: Override Scala version in our build

### PR DESCRIPTION
**Pull request by purplefox**
### Description 

The build is failing with `java.lang.NoClassDefFoundError: scala/$less$colon$less`

This because the build is pulling in different Scala versions from our dependencies.

We depend on `org.apache.kafka:kafka_2.13:jar:6.0.0-ccs-SNAPSHOT` and it has been recently upgraded to Scala 2.13.2

However `io.confluent.ksql:ksqldb-common` depends on `io.confluent:kafka-schema-registry-client:jar:6.0.0-SNAPSHOT` which indirectly depends on scala 2.13.1

This results in two different versions of Scala on the classpath and the build failing.

`2.13.1` and `2.13.2` are binary compatible so it's safe to explicitly declare dependencies on Scala `2.13.2` from our build to override derived versions and make sure there is only one version of Scala on the classpath.

This is the workaround in this PR, that should unbreak the build.

I think a better solution would be for schema-registry to explicitly lock to Scala `2.13.2` to prevent version from it's dependencies (`2.13.1`) from being dragged in.

### Testing done 

Non functional change.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -433,6 +434,20 @@
                 <version>${icu.version}</version>
             </dependency>
 
+            <!-- We need to fix our Scala versions as otherwise our depdendencies bring in
```
Comment (vcrfxia) [inline pom.xml:437]: nit: can we update the comment to make it more clear that this is a temporary workaround?

**Comment by ijuma** [comment]
@purplefox how does schema registry client depend on 2.13.1? That seems weird.

**Comment by purplefox** [comment]
> @purplefox how does schema registry client depend on 2.13.1? That seems weird.

It's not, I misread the dependency tree. It's actually coming in through this route:

```
[INFO] +- io.confluent:kafka-json-schema-provider:jar:6.0.0-SNAPSHOT:compile
[INFO] |  +- com.kjetland:mbknor-jackson-jsonschema_2.13:jar:1.0.39:compile
[INFO] |     +- org.scala-lang:scala-library:jar:2.13.1:compile
```

File: pom.xml [hunk 2]
```diff
@@ -433,6 +434,21 @@
...
+                <groupId>org.scala-lang</groupId>
+                <artifactId>scala-library</artifactId>
+                <version>${scala.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>org.scala-lang</groupId>
+                <artifactId>scala-reflect</artifactId>
+                <version>${scala.version}</version>
+            </dependency>
```
Comment (ijuma) [inline pom.xml:450]: Can we add this to `common` instead? KSQL and Schema Registry both seem to depend on a Scala library now (`mbknor-jackson-jsonschema`), so we should set this in the `dependencyManagement` section of `common`. And the scala version should be kept in sync with the one in Kafka.

**Pull request merged**